syntax = "proto3";
service Greeter {
  rpc UploadFile(stream UploadFileRequest) returns (StringResponse) {}
  rpc DownloadFile(MetaData) returns (stream FileResponse) {}
}

message StringResponse {
  string message = 1;
}
// Download file
message FileResponse{
    bytes chunk_data = 1;
}
message MetaData {
    string filename = 1;
    string extension = 2;
}
message UploadFileRequest {
    oneof request{
        MetaData metadata = 1;
        bytes chunk_data = 2;
    }
}